### ISSUE TITLE
IndexError: list index out of range (in like_history)

### ISSUE THREAD

**Issue by joshinils** [A3M4/YouTube-Report#29]
since when i use `python`i get this same error with the package _tkinter: https://github.com/A3M4/YouTube-Report/issues/21,
i tried using `python3`, but alas:
<pre><font color="#55FF55"><b>nils@niels</b></font>:<font color="#5555FF"><b>~/YouTube-Report</b></font>$ python3 report.py
Could not parse comments.
Traceback (most recent call last):
  File &quot;report.py&quot;, line 33, in &lt;module&gt;
    like, all_likes = HTML().like_history()
  File &quot;/home/nils/YouTube-Report/parse.py&quot;, line 107, in like_history
    link = r&quot;https://www.youtube.com/watch?v=&quot; + match_list[-1][11:]
IndexError: list index out of range
</pre>

i have made sure to downlaod the report in english, that was a previous "mistake"

**Comment by MartinThoma** [comment]
This part does not exist anymore. Could you please pull the latest master, check again and report the new bug? Or close the ticket if it's fixed.

**Comment by joshinils** [comment]
yes, it still has the same error; I wasn't sure anymore  where i cloned it from, but its from here;
<pre><font color="#55FF55"><b>nils@niels</b></font>:<font color="#5555FF"><b>~/YouTube-Report</b></font>$ git remote get-url origin
https://github.com/A3M4/YouTube-Report
<font color="#55FF55"><b>nils@niels</b></font>:<font color="#5555FF"><b>~/YouTube-Report</b></font>$ git pull
remote: Enumerating objects: 4, done.
remote: Counting objects: 100% (4/4), done.
remote: Total 14 (delta 4), reused 4 (delta 4), pack-reused 10
Entpacke Objekte: 100% (14/14), Fertig.
Von https://github.com/A3M4/YouTube-Report
   54e3f5f..943bf20  master     -&gt; origin/master
Aktualisiere 54e3f5f..943bf20
Fast-forward
 parse.py  | 145 <font color="#00AA00">++++++++++++++++++++++++</font><font color="#AA0000">---------------------------------------------------------------</font>
 report.py |   2 <font color="#00AA00">+</font><font color="#AA0000">-</font>
 2 files changed, 41 insertions(+), 106 deletions(-)
<font color="#55FF55"><b>nils@niels</b></font>:<font color="#5555FF"><b>~/YouTube-Report</b></font>$ python3 report.py 
Could not parse comments.
Traceback (most recent call last):
  File &quot;report.py&quot;, line 33, in &lt;module&gt;
    like, all_likes = HTML().like_history()
  File &quot;/home/nils/YouTube-Report/parse.py&quot;, line 88, in like_history
    link = r&quot;https://www.youtube.com/watch?v=&quot; + match_list[-1][11:]
IndexError: list index out of range
</pre>

**Comment by joshinils** [comment]
after installing `python-tk`, i can now run this with python2, but i still get an error:
<pre><font color="#55FF55"><b>nils@niels</b></font>:<font color="#5555FF"><b>~/YouTube-Report</b></font>$ python report.py 
Traceback (most recent call last):
  File &quot;report.py&quot;, line 20, in &lt;module&gt;
    from parse import HTML
  File &quot;/home/nils/YouTube-Report/parse.py&quot;, line 16, in &lt;module&gt;
    class HTML:
  File &quot;/home/nils/YouTube-Report/parse.py&quot;, line 17, in HTML
    html_watch = open(watch_history, &quot;r&quot;, encoding=&quot;utf-8&quot;).read()
TypeError: &apos;encoding&apos; is an invalid keyword argument for this function
</pre>

with python3 it's still the same as above
